app/models/client.rb

```diff
@@ -52,11 +52,11 @@ class Client < ApplicationRecord
   scope :with_inactive_assignment, (-> { with_assignment.merge(Assignment.ended) })
 
   scope :without_assignment, lambda {
-    left_outer_joins(:assignment).where(assignments: { id: nil })
+    left_outer_joins(:assignments).where(assignments: { id: nil })
   }
 
   scope :with_suggested_active_assignment, lambda {
-    joins(:assignment)
+    joins(:assignments)
```

Comment:
You could use `with_assignment` here and in other places (and maybe rename it to plural as well)